src/extrinsic/staking.rs

```diff
@@ -16,235 +16,224 @@
-	}
+		targets: Vec<Self::Address>,
+	) -> ExtrinsicFor<Self, NominateCall<Self::Address>>;
 
 	/// Stop nominating por validating. Effects take place in the next era
-	pub fn staking_chill(
-		&self,
-	) -> StakingChillXt<Signer::ExtrinsicAddress, Signer::Signature, Params::SignedExtra> {
-		compose_extrinsic!(self, STAKING_MODULE, STAKING_CHILL)
-	}
+	fn staking_chill(&self) -> ExtrinsicFor<Self, ChillCall>;
 
 	/// (Re-)set the controller of the stash
 	/// Effects will be felt at the beginning of the next era.
 	/// Must be Signed by the stash, not the controller.
-	pub fn staking_set_controller(
+	fn staking_set_controller(
 		&self,
-		controller: Signer::ExtrinsicAddress,
-	) -> StakingSetControllerXt<Signer::ExtrinsicAddress, Signer::Signature, Params::SignedExtra> {
-		compose_extrinsic!(self, STAKING_MODULE, STAKING_SET_CONTROLLER, controller)
-	}
+		controller: Self::Address,
+	) -> ExtrinsicFor<Self, SetControllerCall<Self::Address>>;
 
 	/// Return the payout call for the given era
-	pub fn payout_stakers(
+	fn payout_stakers(
 		&self,
 		era: u32,
-		account: Runtime::AccountId,
-	) -> StakingPayoutStakersXt<
-		Signer::ExtrinsicAddress,
-		Signer::Signature,
-		Params::SignedExtra,
-		Runtime::AccountId,
-	> {
-		let value = PayoutStakers { validator_stash: account, era };
-		compose_extrinsic!(self, STAKING_MODULE, PAYOUT_STAKERS, value)
-	}
+		account: Self::AccountId,
+	) -> ExtrinsicFor<Self, PayoutStakersCall<Self::AccountId>>;
 
 	/// For New Era at the end of Next Session.
-	pub fn force_new_era(
+	fn force_new_era(&self) -> ExtrinsicFor<Self, ForceNewEraCall>;
+
+	/// Force there to be a new era at the end of sessions indefinitely.
+	fn force_new_era_always(&self) -> ExtrinsicFor<Self, ForceNewEraAlwaysCall>;
+
+	/// Force there to be no new eras indefinitely.
+	fn force_no_era(&self) -> ExtrinsicFor<Self, ForceNewEraAlwaysCall>;
+
+	/// Re-set the payment target for a controller.
+	fn set_payee(&self, payee: Self::Address) -> ExtrinsicFor<Self, SetPayeeCall<Self::Address>>;
+
+	/// Sets the number of validators.
+	fn set_validator_count(&self, count: u32) -> ExtrinsicFor<Self, SetValidatorCountCall>;
+}
+
+impl<Signer, Client, Params, Runtime> CreateStakingExtrinsic
+	for Api<Signer, Client, Params, Runtime>
+where
+	Signer: SignExtrinsic<Runtime::AccountId>,
+	Client: Request,
+	Params: ExtrinsicParams<Runtime::Index, Runtime::Hash>,
+	Runtime: StakingConfig,
+	Compact<Runtime::CurrencyBalance>: Encode,
+{
+	type Balance = Runtime::CurrencyBalance;
+	type RewardDestination = RewardDestination<Self::Address>;
+	type AccountId = Runtime::AccountId;
+
+	fn staking_bond(
 		&self,
-	) -> StakingForceNewEraXt<Signer::ExtrinsicAddress, Signer::Signature, Params::SignedExtra> {
-		compose_extrinsic!(self, STAKING_MODULE, FORCE_NEW_ERA, ForceEra {})
+		controller: Self::Address,
+		value: Self::Balance,
+		payee: Self::RewardDestination,
+	) -> ExtrinsicFor<Self, BondCall<Self::Address, Self::Balance>> {
```

Comment:
I think `ExtrinsicFor` makes it unnecessary complex, since we can use generic associated types now, I suggest:

```rust
impl StakingExtrinsic ...{
       type Extrinsic<Call> = UnchecheckedExtrinsic<Address, Call, Signature, SignedExtra>,
       
       
       fn staking_bond(...) -> Self::Extrinsic<BondCall<...>>
}

```

Reply:
Oh! Awesome. I was actually trying that at some point, but it didn't work out. That's the much better solution, thanks for that ! ❤️ 😍 ❤️

Reply:
🌞  	💞 💕💗💖 🌟💫✨ 🙀  	😍  🌋💝 🎆 🎇 🎉 🎈 🎊  📣 💎 🏆 👑 🌈 🎠 🌄 💯 💮 ‼️ 💲 🌅 ⛲

Reply:
And it works just like that. Wonderful 🦄 🌈 🎠💯✨